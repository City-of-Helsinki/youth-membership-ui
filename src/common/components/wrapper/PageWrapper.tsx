import React, { PropsWithChildren } from 'react';
import { Helmet } from 'react-helmet-async';
import { useTranslation } from 'react-i18next';
import { useLocation } from 'react-router';
import { useSelector } from 'react-redux';

import { isAuthenticatedSelector } from '../../../domain/auth/redux';

type Props = PropsWithChildren<{}>;

function PageWrapper(props: Props) {
  const { t } = useTranslation();
  const { pathname } = useLocation();
  const user = useSelector(isAuthenticatedSelector);

  const getPageTitle = () => {
    switch (pathname) {
      case '/':
        if (user) return 'membershipInformation.pageTitle';
        else return 'login.pageTitle';
      case '/create':
        return 'registration.pageTitle';
      case '/membership-details':
        return 'membershipDetails.title';
      case '/edit':
        return 'edit.pageTitle';
      default:
        return 'appName';
    }
  };

  return (
    <>
      <Helmet>
        <title>{t(getPageTitle())}</title>
        <meta name="description" content={t('meta.description')} />
        <meta property="og:title" content={t(getPageTitle())} />
        <meta property="og:description" content={t('meta.description')} />
      </Helmet>
      {props.children}
    </>
  );
}

export default PageWrapper;
