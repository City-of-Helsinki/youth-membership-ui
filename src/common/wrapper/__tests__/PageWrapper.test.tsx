import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import PageWrapper from '../PageWrapper';

it('matches snapshot', () => {
  const wrapper = shallow(<PageWrapper />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
