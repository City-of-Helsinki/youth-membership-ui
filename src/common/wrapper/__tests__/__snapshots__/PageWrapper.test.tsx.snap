// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<HelmetProvider
  context={Object {}}
>
  <Helmet
    defer={true}
    encodeSpecialCharacters={true}
  >
    <html
      lang="fi"
    />
    <title>
      Nuorisopalvelut
    </title>
    <meta
      content="Jäsenkortilla pääset mukaan nuorisotalojen toimintaan. Jässärin omistajana voit viettää iltaa discossa tai yökahvilassa, hengailla nuorisotalolla, harjoitella bändikämpillä, käydä dj-kurssin, harrasta tanssia, teatteria tai kädentaitoja, hakea projektirahaa, osallistua kesäleireille tai vaikka tuottaa oman tapahtuman."
      name="description"
    />
    <meta
      content="Nuorisopalvelut"
      property="og:title"
    />
    <meta
      content="Jäsenkortilla pääset mukaan nuorisotalojen toimintaan. Jässärin omistajana voit viettää iltaa discossa tai yökahvilassa, hengailla nuorisotalolla, harjoitella bändikämpillä, käydä dj-kurssin, harrasta tanssia, teatteria tai kädentaitoja, hakea projektirahaa, osallistua kesäleireille tai vaikka tuottaa oman tapahtuman."
      property="og:description"
    />
  </Helmet>
</HelmetProvider>
`;
